src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1601 @@
+    value = vertex.properties.GetProperty(property);
+    delta = vertex.delta;
+  }
+  ApplyDeltasForRead(transaction, delta, view, [&deleted, &value, property](const Delta &delta) {
+    switch (delta.action) {
+      case Delta::Action::SET_PROPERTY: {
+        if (delta.property.key == property) {
+          value = delta.property.value;
+        }
+        break;
+      }
+      case Delta::Action::DELETE_DESERIALIZED_OBJECT:
+      case Delta::Action::DELETE_OBJECT:
+      case Delta::Action::RECREATE_OBJECT: {
+        deleted = false;
+        break;
+      }
+      case Delta::Action::ADD_LABEL:
+      case Delta::Action::REMOVE_LABEL:
+      case Delta::Action::ADD_IN_EDGE:
+      case Delta::Action::ADD_OUT_EDGE:
+      case Delta::Action::REMOVE_IN_EDGE:
+      case Delta::Action::REMOVE_OUT_EDGE:
+        break;
+    }
+  });
+  if (deleted) {
+    return {};
+  }
+  return value;
+}
+
+bool HasVertexProperty(const Vertex &vertex, PropertyId property, Transaction *transaction, View view) {
+  return !GetVertexProperty(vertex, property, transaction, view).IsNull();
+}
+
+bool HasVertexEqualPropertyValue(const Vertex &vertex, PropertyId property_id, PropertyValue property_value,
+                                 Transaction *transaction, View view) {
+  return GetVertexProperty(vertex, property_id, transaction, view) == property_value;
+}
+
+bool IsPropertyValueWithinInterval(const PropertyValue &value,
+                                   const std::optional<utils::Bound<PropertyValue>> &lower_bound,
+                                   const std::optional<utils::Bound<PropertyValue>> &upper_bound) {
+  if (lower_bound && (!PropertyValue::AreComparableTypes(value.type(), lower_bound->value().type()) ||
+                      value < lower_bound->value() || (lower_bound->IsExclusive() && value == lower_bound->value()))) {
+    return false;
+  }
+  if (upper_bound && (!PropertyValue::AreComparableTypes(value.type(), upper_bound->value().type()) ||
+                      value > upper_bound->value() || (upper_bound->IsExclusive() && value == upper_bound->value()))) {
+    return false;
+  }
+  return true;
+}
+
+}  // namespace
+
+void DiskStorage::LoadTimestampIfExists() {
+  if (utils::DirExists(config_.disk.durability_directory)) {
+    if (auto last_timestamp_ = durability_kvstore_->Get(lastTransactionStartTimeStamp); last_timestamp_.has_value()) {
+      timestamp_ = std::stoull(last_timestamp_.value());
+    }
+  }
+}
+
+void DiskStorage::LoadIndexInfoIfExists() const {
+  if (utils::DirExists(config_.disk.durability_directory)) {
+    LoadLabelIndexInfoIfExists();
+    LoadLabelPropertyIndexInfoIfExists();
+  }
+}
+
+void DiskStorage::LoadLabelIndexInfoIfExists() const {
+  if (auto label_index = durability_kvstore_->Get(label_index_str); label_index.has_value()) {
+    auto *disk_label_index = static_cast<DiskLabelIndex *>(indices_.label_index_.get());
+    std::vector<std::string> labels = utils::Split(label_index.value(), "|");
```

Comment:
```suggestion
    std::vector<std::string> labels{utils::Split(label_index.value(), "|")};
```
I think this is also better without copy assignment